spirv_reflect.cpp

```diff
@@ -651,7 +651,7 @@ void CompilerReflection::emit_specialization_constants()
 		return;
 
 	json_stream->emit_json_key_array("specialization_constants");
-	for (const auto spec_const : specialization_constants)
+	for (const auto& spec_const : specialization_constants)
```

Comment:
Why? Also, formatting, & binds to right.

Reply:
`-Wrange-for-analysis` complains because `decltype(spec_const)` is a structure with more than one member, so it is more efficient to use `const &`